include/s3select_functions.h

```diff
@@ -1190,6 +1198,101 @@ struct _fn_nullif : public base_function {
       }
     };
 
+struct _fn_when_than : public base_function {
+
+  value when_value,than_value;
+
+  bool operator()(bs_stmt_vec_t* args, variable* result)
+  {
+    bs_stmt_vec_t::iterator iter = args->begin();
+
+    base_statement* than_expr = *iter;
+    iter ++;
+
+    base_statement* when_expr = *iter;
+
+    when_value = when_expr->eval();
+    
+    if (when_value.i64())//true
+    {
+        *result = than_expr->eval();
+        return true;
+    }
+
+    result->set_null();
+
+    return true;
+  }
+};
+
+struct _fn_case_when_else : public base_function {
+
+  value when_than_value;
+
+  bool operator()(bs_stmt_vec_t* args, variable* result)
+  {
+    base_statement* else_expr = *(args->begin());
+
+    size_t args_size = args->size() -1;
+
+    for(int ivec=args_size;ivec>0;ivec--)
+    {
+      when_than_value = (*args)[ivec]->eval();
+      
+      if(!when_than_value.is_null())
+      {
+        *result = when_than_value;
+        return true;
+      }
+
+    }
+
+    *result = else_expr->eval();
+    return true;
+  }
+};
+
+struct _fn_coalesce : public base_function
+{
+
+  value res;
+
+  bool operator()(bs_stmt_vec_t* args, variable* result)
+  {
+    bs_stmt_vec_t::iterator iter_begin = args->begin();
+    int args_size = args->size();
+    while (args_size >= 1)
+    {
+      base_statement* expr = *iter_begin;
+      value expr_val = expr->eval();
+      iter_begin++;
+      if ( !(expr_val.is_null())) {
+          *result = expr_val;
+          return true;
+        } 
+      args_size--;
+    }
+    result->set_null();
+    return true;
+  }
+};
+
+struct _fn_cast : public base_function
+{
+
```

Comment:
the name /cast/ is wrong naming , and also the implementation
cast to string, means converting other types (double,integer,timestamp) into string presentation.

the operator is only evaluate the expression , it's not converting it to string